base_layer/core/src/block.rs

```diff
@@ -157,4 +166,135 @@ impl From<Transaction> for AggregateBody {
     }
 }
 
+#[derive(Default)]
+pub struct BlockBuilder {
+    pub header: Option<BlockHeader>,
```

Comment:
Given that you allow for a blank header anyway, just start one and drop the option.